packages/plugin-birdeye/src/actions/defi/get-ohlcv.ts

```diff
@@ -0,0 +1,387 @@
+                type: "15m",
+            },
+        ],
+    },
+};
+
+type OHLCVResponse = typeof exampleResponse;
+
+type TimeInterval =
+    | "1m"
+    | "3m"
+    | "5m"
+    | "15m"
+    | "30m"
+    | "1H"
+    | "2H"
+    | "4H"
+    | "6H"
+    | "8H"
+    | "12H"
+    | "1D"
+    | "3D"
+    | "1W"
+    | "1M";
+
+const TIME_INTERVALS: Record<TimeInterval, string[]> = {
+    "1m": ["1 minute", "1min", "1m"],
+    "3m": ["3 minutes", "3min", "3m"],
+    "5m": ["5 minutes", "5min", "5m"],
+    "15m": ["15 minutes", "15min", "15m"],
+    "30m": ["30 minutes", "30min", "30m"],
+    "1H": ["1 hour", "1hr", "1h"],
+    "2H": ["2 hours", "2hr", "2h"],
+    "4H": ["4 hours", "4hr", "4h"],
+    "6H": ["6 hours", "6hr", "6h"],
+    "8H": ["8 hours", "8hr", "8h"],
+    "12H": ["12 hours", "12hr", "12h"],
+    "1D": ["1 day", "daily", "1d"],
+    "3D": ["3 days", "3day", "3d"],
+    "1W": ["1 week", "weekly", "1w"],
+    "1M": ["1 month", "monthly", "1m"],
+};
+
+const DEFAULT_INTERVAL: TimeInterval = "1D";
+
+// Constants for keyword matching
+const OHLCV_KEYWORDS = [
+    "ohlc",
+    "ohlcv",
+    "candlestick",
+    "candle",
+    "chart",
+    "price history",
+    "open close high low",
+    "opening price",
+    "closing price",
+    "historical",
+] as const;
+
+// Helper function to check if text contains OHLCV-related keywords
+const containsOHLCVKeyword = (text: string): boolean => {
+    return OHLCV_KEYWORDS.some((keyword) =>
+        text.toLowerCase().includes(keyword.toLowerCase())
+    );
+};
+
+const extractTimeInterval = (text: string): TimeInterval => {
+    const lowerText = text.toLowerCase();
+
+    // First try exact matches
+    for (const [interval, keywords] of Object.entries(TIME_INTERVALS)) {
+        if (keywords.some((keyword) => lowerText.includes(keyword))) {
+            return interval as TimeInterval;
+        }
+    }
+
+    // Then try common variations
+    if (lowerText.includes("hourly")) return "1H";
+    if (lowerText.includes("daily")) return "1D";
+    if (lowerText.includes("weekly")) return "1W";
+    if (lowerText.includes("monthly")) return "1M";
+
+    return DEFAULT_INTERVAL;
+};
+
+const formatVolume = (volume: number): string => {
+    if (volume >= 1_000_000_000) {
+        return `$${(volume / 1_000_000_000).toFixed(2)}B`;
+    }
+    if (volume >= 1_000_000) {
+        return `$${(volume / 1_000_000).toFixed(2)}M`;
+    }
+    if (volume >= 1_000) {
+        return `$${(volume / 1_000).toFixed(2)}K`;
+    }
+    return `$${volume.toFixed(2)}`;
+};
+
+const getOHLCVData = async (
+    apiKey: string,
+    contractAddress: string,
+    chain: Chain,
+    interval: TimeInterval = DEFAULT_INTERVAL
+): Promise<OHLCVResponse | null> => {
+    try {
+        const params = new URLSearchParams({
+            address: contractAddress,
+            interval: interval.toLowerCase(),
+            limit: "24", // Get last 24 periods
+        });
```

Comment:
These don't appear to be the correct params for this endpoint.  
The required params for v1 are:
- address
- type
- time_from
- time_to 
According to the [documentation](https://docs.birdeye.so/reference/get_defi-ohlcv). I tested and verified in my branch.

```
        // Get current unix time and 24h ago
        const endUnixTime = Math.floor(Date.now() / 1000);
        const startUnixTime = endUnixTime - (60 * 60 * 24); // 24 hours ago

        const params = new URLSearchParams({
            address: contractAddress,
            // interval: interval.toLowerCase(),
            type: "1D", // Get last 24 periods
            time_from: startUnixTime.toString(),
            time_to: endUnixTime.toString(),
 ``` 
 
 
![image](https://github.com/user-attachments/assets/8ab29de8-2770-4614-a54d-76f62b50c1ef)

Reply:
Thanks, @camharris . I have several big changes coming soon that will address this. Thanks for your review I appreciate it.